stl/inc/limits

```diff
@@ -1009,24 +1011,96 @@ public:
     static constexpr int min_exponent10 = LDBL_MIN_10_EXP;
 };
 
-#ifdef __cpp_lib_bitops // TRANSITION, VSO-1020212
+// Implementation of countl_zero without using specialized CPU instructions.
+// Used at compile time and when said instructions are not supported.
+// see "Hacker's Delight" section 5-3
 template <class _Ty>
-inline constexpr bool _Is_standard_unsigned_integer =
-    _Is_any_of_v<remove_cv_t<_Ty>, unsigned char, unsigned short, unsigned int, unsigned long, unsigned long long>;
+_NODISCARD constexpr int _Countl_zero_helper(_Ty _Val) noexcept {
+    _Ty _Yy = 0;
+
+    unsigned int _Nn = numeric_limits<_Ty>::digits;
+    unsigned int _Cc = numeric_limits<_Ty>::digits / 2;
+    do {
+        _Yy = static_cast<_Ty>(_Val >> _Cc);
+        if (_Yy != 0) {
+            _Nn  = _Nn - _Cc;

```

Comment:
Would compound assignment be simpler?

```suggestion
            _Nn  -= _Cc;
```

Reply:
I think the reason I didn't do that was just to mirror what was in hacker's delight, I have no problem with -= though